src/includes/load-service.h

```diff
@@ -1544,6 +1662,87 @@ void process_service_line(settings_wrapper &settings, const char *name, const ch
             settings.run_in_cgroup = read_setting_value(input_pos, i, end, nullptr);
             break;
         #endif
+        #if SUPPORT_CAPABILITIES
+        case setting_id_t::CAPABILITIES:
+            read_setting_value(settings.capabilities, setting_op, input_pos, i, end, nullptr, ',');
+            break;
+        case setting_id_t::SECURE_BITS:
+        {
+            std::list<std::pair<unsigned,unsigned>> indices;
+            string onstart_cmds = read_setting_value(input_pos, i, end, &indices);
+            // plain assignment will clear, while append will add more
+            if (setting_op != setting_op_t::PLUSASSIGN) {
+                settings.secbits.clear();
+            }
+            for (auto indexpair : indices) {
+                string secbit_txt = onstart_cmds.substr(indexpair.first,
+                        indexpair.second - indexpair.first);
+                if (secbit_txt == "keep-caps") {
+                    settings.secbits.keep_caps = true;
+                }
+                else if (secbit_txt == "keep-caps-locked") {
+                    settings.secbits.keep_caps_locked = true;
+                }
+                else if (secbit_txt == "no-setuid-fixup") {
+                    settings.secbits.no_setuid_fixup = true;
+                }
+                else if (secbit_txt == "no-setuid-fixup-locked") {
+                    settings.secbits.no_setuid_fixup_locked = true;
+                }
+                else if (secbit_txt == "noroot") {
+                    settings.secbits.noroot = true;
+                }
+                else if (secbit_txt == "noroot-locked") {
+                    settings.secbits.noroot_locked = true;
+                }
+                else {
+                    throw service_description_exc(name, "unknown secure bit: " + secbit_txt,
+                            "secure-bits", input_pos);
+                }
+            }
+            break;
+        }
+        #endif
+        case setting_id_t::NICE:
+        {
+            string nice_str = read_setting_value(input_pos, i, end);
+            settings.nice = 1 | ((int)parse_snum_param(input_pos, nice_str, name,
```

Comment:
the reason i didn't do that is that 1) it actually makes the code more complicated because now you have to check a bunch of booleans around the place, particularly when copying things between the structures and 2) it adds a bunch of pointless structural bloat due to field alignment, and when you take care to put the booleans in one place, it becomes less obvious how they are associated with the rest of the fields

Reply:
There should only be a couple of extra assignments I would think? You don't have to check the booleans, just pass them around along with the nice value. I know what you mean about putting the booleans in one place separate from the other fields they may be related to but it doesn't seem like that much of a deal and is ultimately a lot cleaner.